satellite/satellitedb/updatechecker.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"sync"
+	"time"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/overlay"
+)
+
+// UpdateChecker is an interface that determines whether updates need to be
+// applied to the underlying overlay cache db.
+type UpdateChecker interface {
+	AddressRequiresUpdate(node *pb.Node) bool
+	GetNodeStats(nodeId storj.NodeID, isUp bool) (stats *overlay.NodeStats)
+	SetNodeStats(nodeId storj.NodeID, isUp bool, stats *overlay.NodeStats)
+}
+
+type addressInfo struct {
+	address   string
+	lastIp    string
+	transport pb.NodeTransport
+}
+
+type uptimeInfo struct {
+	isUp       bool
+	lastUptime time.Time
+	stats      *overlay.NodeStats
+}
+
+type CachedChecker struct {
+	addressLock  sync.RWMutex
+	addressCache map[storj.NodeID]*addressInfo
+
+	uptimeLock  sync.RWMutex
+	uptimeCache map[storj.NodeID]*uptimeInfo
+	isUpDur     time.Duration
+}
+
+func NewCachedChecker(isUpDur time.Duration) *CachedChecker {
+	return &CachedChecker{
+		addressCache: make(map[storj.NodeID]*addressInfo),
+		uptimeCache:  make(map[storj.NodeID]*uptimeInfo),
+		isUpDur:      isUpDur,
+	}
+}
+
+// AddressRequiresUpdate returns true if the address should be updated in the
+// underlying db, and false if not. It also keeps the internal cache up-to-date.
+func (c *CachedChecker) AddressRequiresUpdate(node *pb.Node) bool {
```

Comment:
This is also modifies the state, however the method name implies it doesn't change it.

Reply:
Fixed.